central/sensorupgradeconfig/datastore/singleton.go

```diff
@@ -35,7 +35,9 @@ func addDefaultConfigIfEmpty(d DataStore) error {
 	if currentConfig != nil {
```

Comment:
I think what you want here is to also check
```suggestion
	if currentConfig != nil && (!env.ManagedCentral.BooleanSetting() || !currentConfig.GetEnableAutoUpgrade()) {
```
because in managed mode, you want to make sure that auto-upgrade is disabled, even if for some reason it is stored as enabled in the DB.

Reply:
That makes sense. Thanks!